osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -111,38 +116,39 @@ static IEnumerable<HitObject> enumerateHitObjects(IEnumerable<HitObject> hitObje
             }
         }
 
-        private void markVisibleDrawableObjectsHit()
+        private void preventMissOnPreviousHitObjects()
         {
-            if (!DrawableRuleset.Playfield.IsLoaded)
+            void preventMiss(HitObject hitObject)
             {
-                Schedule(markVisibleDrawableObjectsHit);
-                return;
-            }
+                if (hitObject.StartTime > editorState.Time)
+                    return;
 
-            foreach (var drawableObject in enumerateDrawableObjects(DrawableRuleset.Playfield.AllHitObjects, editorState.Time))
-            {
-                if (drawableObject.Entry == null)
-                    continue;
+                var drawableObject = DrawableRuleset.Playfield.HitObjectContainer
+                                                    .AliveObjects
+                                                    .LastOrDefault(it => it.HitObject == hitObject);
```

Comment:
have not checked but am very skeptical this is going to handle nested drawable hit objects correctly